simtools/configuration/configurator.py

```diff
@@ -288,6 +288,20 @@ def _initialize_io_handler(self):
             model_path=self.config.get("model_path", None),
         )
 
+    def _initialize_output(self):
+        """
+        Initialize default output file names (in out output_file is not configured).
+        """
+        if self.config.get("output_file", None) is None:
+            if self.config.get("test", False):
+                self.config["output_file"] = "TEST"
```

Comment:
I see your point and I agree that this is a bit difficult to read. But not sure how to do it here with prefix / labels, given that we still have to add all the if statements. 

Note it is e.g., `TEST.yaml` (this would be only f"{prefix}.{extension}"), but f"{prefix}-{label}.{extension}" otherwise (would be another if). Also in some cases a output_file is given with extensions, sometimes without. I think using an f-string would actually complicate this code.

Reply:
This is my (untested!) suggestion:
```
def _initialize_output(self):
    """
    Initialize default output file names (in out output_file is not configured).
    """
    if self.config.get("output_file", None) is None:
        prefix = "TEST"
        label = extention = ""
        if not self.config.get("test", False):
            prefix = self.config["activity_id"]
            if self.config.get("label", "") and len(self.config.get("label", "")) > 0:
                label = f"-{self.config['label']}"
        if len(self.config.get("output_file_format", "")) > 0:
            extention = f".{self.config['output_file_format']}"

        self.config["output_file"] = f"{prefix}{label}{extention}"  # Or return this string and assign it to the class attribute externally
```

Again, this is really not critical, just a personal preference and I am not 100% that it is clearer. So feel free to keep as is.

Reply:
OK, I changed it (unit tests are ok on your code, so I assume it doing what it is supposed to do).